lms/static/scripts/frontend_apps/components/dashboard/OrderableActivityTable.tsx

```diff
@@ -46,7 +47,13 @@ export default function OrderableActivityTable<T>({
       columns.map(({ field, label }, index) => ({
         field,
         label,
-        classes: index === 0 ? 'lg:w-[60%] md:w-[45%]' : undefined,
+        classes: classnames({
+          // For assignments with auto-grading, a fifth column is displayed.
+          // In that case, we need to reserve less space for the first column,
+          // otherwise the rest overflow.
+          'lg:w-[60%] md:w-[45%]': index === 0 && columns.length < 5,
+          'lg:w-[45%] md:w-[30%]': index === 0 && columns.length >= 5,
```

Comment:
This is starting to get coupled with concrete usages of the component, so we may have to extract it from here.